language_formatters_pre_commit_hooks/pretty_format_java.py

```diff
@@ -128,7 +185,52 @@ def pretty_format_java(argv: typing.Optional[typing.List[str]] = None) -> int:
                 ", ".join(output.splitlines()),
             ),
         )
+    return 0 if status == 0 else 1
+
+
+def format_palantir(args):
+    # Palantir Java Formatter starting from 2.0.0 does not support Java 1.8,
+    # before that version the tool can only be executed on Java up to version 20.
+    # Context: https://github.com/palantir/palantir-java-format/releases/tag/v2.0.0
+    if Version(args.palantir_java_formatter_version) >= Version("2.0.0"):
+        assert_min_jdk_version(Version("11.0"))  # pragma: no cover
+
+    # Palantir Java formatter is not compatible with Java 21 and above yet.
+    # https://github.com/palantir/palantir-java-format/issues/977 contains
+    # more links for Java compatibility
+    assert_max_jdk_version(Version("21.0"))  # pragma: no cover
+
+    if args.palantir_java_formatter_jar is None:
+        palantir_java_formatter_jar = _download_palantir_java_formatter_jar(
+            args.palantir_java_formatter_version,
+        )
+    else:
+        palantir_java_formatter_jar = args.palantir_java_formatter_jar
+
+    cmd_args = [
+        "java",
+        "-jar",
```

Comment:
Palantir Java Formatter has additional permissions inside manifest and they aren't required to be provided separately